docs/source/fx.rst

```diff
@@ -298,6 +290,209 @@ Miscellanea
+    class SubclassM(M):
+        def __init__(self):
+            super().__init__()
+
+        # Paste the generated `forward` function (the one we printed and
+        # copied on line 22) here
+        def forward(self, y):
+            x = self.x
+            add_1 = x + y;  x = y = None
+            return add_1
+
+    # Create an instance of the original, untraced Module. Then, create an
+    # instance of the Module with the copied `forward` function. We can
+    # now compare the output of both the original and the traced version.
+    pre_trace = M()
+    post_trace = SubclassM()
+
+
+Use the ``to_folder`` Function From ``GraphModule``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:meth:`GraphModule.to_folder` is a method in ``GraphModule`` that allows
+you to dump out the generated FX code to a folder. Although copying the
+forward pass into the code often suffices as in :ref:`Print the Generated Code`,
+it doesn’t capture any model attribute state. To examine modules and parameters,
+we can use ``to_folder``.
+
+::
+
+    m = symbolic_trace(M())
+    m.to_folder("foo", "Bar")
+    from foo import Bar
+    y = Bar()
+
+After running the above example, we can then look at the code within
+``foo/module.py`` and modify it as desired to debug the generated code.
+
+Debugging the Transformation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First, read the FX
+documentation. Understand the most important classes (:class:`Node`,
+:class:`Graph`, :class:`Proxy`, and :class:`Tracer`), and determine which
+class attributes might give you the best representation of your
+program’s intermediate state.
+
+There are several ways to examine what ``symbolic_tracing`` produces:
+
+::
+
+    # Sample Module
+    class M(torch.nn.Module):
+        def forward(self, x, y):
+            return x + y
+
+    # Create an instance of `M`
+    m = M()
+
+    # Symbolically trace an instance of `M` (returns a GraphModule)
+    traced = symbolic_trace(m)
+
+    # Print the code produced by tracing the module. The generated `forward`
+    # function is:
+    """
+    def forward(self, x, y):
+        add_1 = x + y;  x = y = None
+        return add_1
+    """
+    print(traced)
+
+    # Print the internal Graph. This representation returns:
+    """
+    graph(x, y):
+        %add_1 : [#users=1] = call_function[target=<built-in function add>](args = (%x, %y), kwargs = {})
+        return add_1
+    """
+    print(traced.graph)
+
+    # Print a tabular representation of the internal Graph. This gives us:
+    """
+    opcode         name    target                   args      kwargs
+    -------------  ------  -----------------------  --------  --------
+    placeholder    x       x                        ()        {}
+    placeholder    y       y                        ()        {}
+    call_function  add_1   <built-in function add>  (x, y)    {}
+    """
+    traced.graph.print_tabular()
+
+Using the above example, let’s say that the call to ``print(traced)``
```

Comment:
This section is titled "Debugging the transformation" but this prose talks about debugging symbolic tracing. I think we should probably focus on the former, as that's where most of the user debugging will happen if we've done our job correctly